### ISSUE TITLE
DashboardPanelShortcuts error with external links

### ISSUE THREAD

**Issue by zoeck-dev** [philippdaun/processwire-dashboard#41]
**Describe the bug**
When I use the shortcuts with external links, I always get an error
PHP Notice: Trying to get property 'url' of non-object in ...\site\modules\Dashboard\DashboardPanelShortcuts.module:226

**To Reproduce**
Add an external Link to the DashboardPanelShortcuts

**Additional context**
I think the check doesn't fit that way (line 226):
if (!$page || !$page->url) {

Because if "$page" does not exist, then "$page->url" will cause an error.

https://processwire.com/talk/topic/22847-processwire-dashboard/page/7/

**Comment by daun** [comment]
Mmh. Not sure. The if statement would just skip the second part if the first part is true.

Can you post how you're setting up the panel itself, what you're passing into the 'shortcuts' option?

**Comment by zoeck-dev** [comment]
Sure :) 

Heres my code:

    $allshortcuts[] = "";
    $allshortcuts['Plan erstellen'] = [$this->config->urls->admin.'page/add/?parent_id=1024&template_id=47', 'calendar',     'Einen neuen Plan erstellen.'];
    $allshortcuts['News eintragen'] = [$this->config->urls->admin.'page/add/?parent_id=1023&template_id=45', 'newspaper-o', 'Hier kann ein neuer News Eintrag erstellt werden.'];
    $allshortcuts['User syncronisieren'] = [$this->config->urls->admin . 'setup/syncuser/', 'refresh', 'User akutalisieren.'];
    $allshortcuts['Updates'] = [$this->config->urls->admin . 'setup/upgrades/', 'download', 'Systemupdates suchen und installieren.'];
    $allshortcuts['Einstellungen'] = [$this->config->urls->admin . 'page/edit/?id=' . $this->pages->get("/verwaltung/einstellungen/")->id, 'cogs', 'Hier können die Einstellungen geändert werden.'];
    $allshortcuts['Alle Seiten'] = ['/ugs/page/', 'files-o', 'Alle vorhandenen Seiten anzeigen.'];

    $panels->add([
        'panel' => 'shortcuts',
        'title' => 'Menü',
        'data' => [
            'shortcuts' => $allshortcuts,
            'display' => 'list',
            'fallbackIcon' => 'star-o',
            'checkAccess' => true
        ]
    ]);

**Comment by szabeszg** [comment]
> The if statement would just skip the second part if the first part is true.

Only if AND is used but no OR, I guess, so 
`if (!$page || !isset($page->url)) {`
should be needed perhaps?

**Comment by szabeszg** [comment]
BTW, I get the same PHP Notice, so it an issue. I modifed the code as I showed above, it work for me with isset().
https://stackoverflow.com/questions/8138269/multiple-boolean-conditions-operator-precedence
"&& depends of the evaluation of the right expression when left one is true, || doesn't."

**Comment by daun** [comment]
`$page` is not an object in this case, so it does exist but doesn't allow property access. @szabeszg's fix should work. I'll try and plug the example config into my test setup to see why `$page` is not an object in the first place.

**Comment by daun** [comment]
I don't think this is about faulty boolean logic. If `$page` is empty, the first condition `!$page` is already `true` so it won't evaluate the second condition since `true || [anything]` will always be `true`.

**Comment by szabeszg** [comment]
> If `$page` is empty, the first condition `!$page` is already `true` so it won't evaluate the second condition since `true || [anything]` will always be `true`.

Apparently it does evaluate the second one...

**Comment by szabeszg** [comment]
Looks like it fails when $page is a string like: '/admin/page/edit/?id=1'

**Comment by daun** [comment]
Yeah, that makes sense — if it's a string, the first condition is false, so it will evaluate the second condition. If it was completely empty, it wouldn't. Your `isset` solution is the way to go here...

**Comment by BernhardBaumrock** [comment]
Just for reference here is the line in question: https://github.com/daun/processwire-dashboard/blob/e514fb6a9e9224f09c34997467b87000d8602a9d/DashboardPanelShortcuts.module#L226

Why would a string find it's way to that line? While isset() would work wouldn't it be better to ensure that $page is a Page object at the very top?

**Comment by BernhardBaumrock** [comment]
eg this (but I don't know how your module works exactly and what syntax you expect...):
```php
if(is_string($page)) $page = $this->wire->pages->get($page);
```

**Comment by daun** [comment]
The string just means it's already a URL. So yeah, I should go with an additional `is_object` or `is_a` and then `isset`. Better safe than sorry.

**Comment by daun** [comment]
Fixed in [1.5.2](https://github.com/daun/processwire-dashboard/releases/tag/v1.5.2), thanks everybody for reporting and helping troubleshoot!
